score-powershell/src/main/java/io/cloudslang/content/services/WSManRemoteShellService.java

```diff
@@ -0,0 +1,397 @@
+        httpClientInputs.setX509HostnameVerifier(wsManRequestInputs.getX509HostnameVerifier());
+        httpClientInputs.setProxyHost(wsManRequestInputs.getProxyHost());
+        httpClientInputs.setProxyPort(wsManRequestInputs.getProxyPort());
+        httpClientInputs.setProxyUsername(wsManRequestInputs.getProxyUsername());
+        httpClientInputs.setProxyPassword(wsManRequestInputs.getProxyPassword());
+        httpClientInputs.setKeystore(wsManRequestInputs.getKeystore());
+        httpClientInputs.setKeystorePassword(wsManRequestInputs.getKeystorePassword());
+        httpClientInputs.setTrustKeystore(wsManRequestInputs.getTrustKeystore());
+        httpClientInputs.setTrustPassword(wsManRequestInputs.getTrustPassword());
+        String headers = httpClientInputs.getHeaders();
+        if (StringUtils.isEmpty(headers)) {
+            httpClientInputs.setHeaders(CONTENT_TYPE_HEADER);
+        } else {
+            httpClientInputs.setHeaders(headers + NEW_LINE_SEPARATOR + CONTENT_TYPE_HEADER);
+        }
+        httpClientInputs.setMethod(HttpPost.METHOD_NAME);
+        return httpClientInputs;
+    }
+
+    /**
+     * This method executes a request with the given ScoreHttpClient, HttpClientInputs and body.
+     *
+     * @param scoreHttpClient
+     * @param httpClientInputs
+     * @param requestMessage
+     * @return the result of the request execution.
+     */
+    private Map<String, String> executeRequest(ScoreHttpClient scoreHttpClient, HttpClientInputs httpClientInputs, String requestMessage) {
+        httpClientInputs.setBody(requestMessage);
+        return scoreHttpClient.execute(httpClientInputs);
+    }
+
+    /**
+     * Creates a shell on the remote server and returns the shell id.
+     *
+     * @param scoreHttpClient
+     * @param httpClientInputs
+     * @param resourceName
+     * @param url
+     * @param wsManRequestInputs
+     * @return the id of the created shell.
+     * @throws RuntimeException
+     * @throws IOException
+     * @throws URISyntaxException
+     * @throws TransformerException
+     * @throws XPathExpressionException
+     * @throws SAXException
+     * @throws ParserConfigurationException
+     */
+    private String createShell(ScoreHttpClient scoreHttpClient, HttpClientInputs httpClientInputs, String resourceName, String url,
+                               WSManRequestInputs wsManRequestInputs) throws RuntimeException, IOException, URISyntaxException,
+            TransformerException, XPathExpressionException, SAXException, ParserConfigurationException {
+        String document = ResourceLoader.loadAsString(resourceName);
+        document = createCreateShellRequestBody(document, url, String.valueOf(wsManRequestInputs.getMaxEnvelopeSize()),
+                wsManRequestInputs.getWinrmLocale(), String.valueOf(wsManRequestInputs.getOperationTimeout()));
+        Map<String, String> createShellResult = executeRequest(scoreHttpClient, httpClientInputs, document);
```

Comment:
the code below this comment seems to be duplicated in the executeCommand method, maybe you could extract it in a new method